bin/arno-iptables-firewall

```diff
@@ -47,28 +47,18 @@ fi
 
 # Check if the environment file exists and if so, load it
 #########################################################
-if [ -z "$ENV_FILE" ]; then
-  if [ -f /usr/local/share/arno-iptables-firewall/environment ]; then
-    ENV_FILE="/usr/local/share/arno-iptables-firewall/environment"
-  else
-    if [ -f /usr/share/arno-iptables-firewall/environment ]; then
-      ENV_FILE="/usr/share/arno-iptables-firewall/environment"
-    else
-      printf "\033[40m\033[1;31mERROR: The environment file (ENV_FILE) has not been specified\033[0m\n" >&2
-      printf "\033[40m\033[1;31m       in the configuration file. Try upgrading your config-file!\033[0m\n\n" >&2
-      exit 2
-    fi
-  fi
-fi
 
-if [ -e "$ENV_FILE" ]; then
-  . "$ENV_FILE"
-else
-  printf "\033[40m\033[1;31mERROR: Could not read environment file \"$ENV_FILE\"!\033[0m\n" >&2
+# Autodetect according to the path of the main script
+ENV_FILE="${0%/*}/../share/arno-iptables-firewall/environment"
```

Comment:
@abelbeck : I've changed/fixed the above. The only alternative I can think of atm is simply detecting for /usr/local/share/environment and (simply) fallback to /usr/share/environment in case it doesn't exist.

Reply:
@arnova : It seems `$(readlink -f "$0")` will return a clean full pathname.

But some say `readlink` is not always installed by default.  I have never used it myself but our Busybox config has it.

Reply:
@arnova : for the main script I kind of like hardcoded autodetect, ie (untested)
```
# Autodetect according to standard paths
unset IFS
for x in /usr/local /usr; do
  ENV_FILE="$x/share/arno-iptables-firewall/environment"
  if [ -f "$ENV_FILE" ]; then
    break
  fi
done
```
replaces:
```
# Autodetect according to the path of the main script
ENV_FILE="${0%/*}/../share/arno-iptables-firewall/environment"
```
This would allow the `arno-iptables-firewall` script to be anywhere, useful for testing, debugging and alternate bin locations.

Reply:
Agreed. I've updated the PR. Is it good to go as is (after squash) ?